crates/storage/src/db.rs

```diff
@@ -0,0 +1,28 @@
+use crate::redb_interface::ReamDB;
+use crate::errors::StoreError;
+
+pub struct Database{
+    /// Database for the ream client 
+    /// 
+    /// Currently this implementation has a single reference to redb instance
+    /// but eventually we would like to add support for multiple backends and 
+    /// can be set behind feature flags
```

Comment:
I don't think we should support multiple backends. For BLS fine that was fairly minimally, but for properly using a db it is a little more involved then a db expessially to get performance or to ensure crashes don't corrupt the database.